socorro/external/boto/connection_context.py

```diff
@@ -0,0 +1,245 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import boto
+import boto.s3.connection
+import boto.exception
+
+import json
+import os
+import socket
+import datetime
+import contextlib
+
+from configman import Namespace, RequiredConfig, class_converter
+from configman.converters import class_converter, py_obj_to_str
+from socorro.lib.util import DotDict
+from socorro.lib.converters import change_default
+
+class BotoNotFound(Exception):
+    pass
+
+
+#==============================================================================
+class ConnectionContext(RequiredConfig):
+
+    required_config = Namespace()
+    required_config.add_option(
+        'host',
+        doc="The hostname (leave empty for AWS)",
+        default="",
+        reference_value_from='resource.boto',
+    )
+    required_config.add_option(
+        'port',
+        doc="The network port (leave at 0 for AWS)",
+        default=0,
+        reference_value_from='resource.boto',
+    )
+    required_config.add_option(
+        'access_key',
+        doc="access key",
+        default="",
+        reference_value_from='resource.boto',
+    )
+    required_config.add_option(
+        'secret_access_key',
+        doc="secret access key",
+        default="",
+        secret=True,
+        reference_value_from='secrets.boto',
+    )
+    required_config.add_option(
+        'bucket_name',
+        doc="The name of the bucket.",
+        default='crashstats',
+        reference_value_from='resource.boto',
+        likely_to_be_changed=True,
+    )
+    required_config.add_option(
+        'prefix',
+        doc="a prefix to use inside the bucket",
+        default='',
+        reference_value_from='resource.boto',
+        likely_to_be_changed=True,
+    )
+    required_config.add_option(
+        'calling_format',
+        doc="fully qualified python path to the boto calling format function",
+        default='boto.s3.connection.SubdomainCallingFormat',
+        from_string_converter=class_converter,
+        reference_value_from='resource.boto',
+        likely_to_be_changed=True,
```

Comment:
because we're specifying a class that is specific to S3 that wouldn't be used if we were connecting to Ceph. so this is an option that is likely to change for other implementations.